### PR TITLE
Step to volmdlr: Arc3D from step to volmdlr for Cylindrical and Toroidal Surface

### PR DESCRIPTION
**Pull request by GabrielJMS**
* **Please check if the PR fulfills these requirements**
- [ ] The commit messages follows our guidelines (explicit, in english)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] CHANGELOG.md has been updated

* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
- [x ] Bug fix:
- [ ] New features:
- [ ] Documentation
- [ ] Optimization

* **Other information**:

# # to be merged after the the holes in #284

### CODE REVIEW

File: scripts/step/read_steps.py [hunk 1]
```diff
@@ -4,36 +4,66 @@
...
 import volmdlr.step
+import volmdlr.faces as vmf
+
 # import volmdlr.cloud as vmcd
 
+
 for step_file in [
-    'tore1.step',
-    'cone1.step',
-    'cone2.step',
```
Comment (masfaraud) [inline scripts/step/read_steps.py:12]: Put back simple steps tests?

File: volmdlr/faces.py [hunk 2]
```diff
@@ -642,40 +708,65 @@ def face_from_contours3d(self,
...
                 delta_x2 = 0.
+
+            delta = last_primitive.end - primitives[0].start
+            new_primitives = []
             for prim in primitives:
-                prim.start.x = abs(self.x_periodicity
-                                   - prim.start.x)
-                prim.end.x = abs(self.x_periodicity
-                                 - prim.end.x)
+                if isinstance(self, SphericalSurface3D):
```
Comment (masfaraud) [inline volmdlr/faces.py:744]: I hate this pattern if isinstance: can we switch to a method that each concept would implement

**Review by masfaraud — CHANGES_REQUESTED** [review]

File: volmdlr/faces.py [hunk 3]
```diff
@@ -1800,21 +2114,92 @@ def fullarc3d_to_2d(self, fullarc3d):
...
+
+        return [vme.BSplineCurve2D.from_points_interpolation(new_points, bspline_curve3d.degree,
+                                                             bspline_curve3d.periodic)]
+        # return [vme.BSplineCurve2D(
+        #     bspline_curve3d.degree,
+        #     control_points=new_points,
+        #     knot_multiplicities=bspline_curve3d.knot_multiplicities,
+        #     knots=bspline_curve3d.knots,
+        #     weights=bspline_curve3d.weights,
+        #     periodic=bspline_curve3d.periodic)]
```
Comment (WirajanDASILVA) [inline volmdlr/faces.py:2183]: clean?

File: volmdlr/faces.py [hunk 4]
```diff
@@ -1800,21 +2114,92 @@ def fullarc3d_to_2d(self, fullarc3d):
...
+        # angle = abs(start.x-end.x)
+        # if arc3d.is_trigo:
+        # end = start + volmdlr.Point2D(arc3d.angle, 0)
+        # else:
+        #     end = start + volmdlr.Point2D(-arc3d.angle, 0)
+        # interior = self.point3d_to_2d(arc3d.interior)
+        # if start.x < interior.x:
+        #     end = start + volmdlr.Point2D(arc3d.angle, 0)
+        # else:
+        #     end = start - volmdlr.Point2D(arc3d.angle, 0)
```
Comment (WirajanDASILVA) [inline volmdlr/faces.py:2130]: clean?

File: volmdlr/faces.py [hunk 5]
```diff
@@ -693,28 +790,36 @@ def repair_primitives_periodicity(self, primitives, last_primitive):
...
+            new_primitives = [prim.translation(delta) for prim in primitives]
+            # temp = prim.translation(delta)
+            # prim.start.y = temp.start.y
+            # prim.end.y = temp.end.y
+
+            # for prim in primitives:
+            #     prim.start.y = abs(self.y_periodicity
+            #                        - prim.start.y)
+            #     prim.end.y = abs(self.y_periodicity
+            #                      - prim.end.y)
```
Comment (WirajanDASILVA) [inline volmdlr/faces.py:803]: clean?

**Comment by GabrielJMS** [comment]
Closing because this PR is outdated and a new one has replaced it

**Pull request closed without merging**